authors/apps/authentication/tests/test_reset_password.py

```diff
@@ -0,0 +1,52 @@
+from rest_framework.views import status
+from django.urls import reverse
+
+from .base_test import BaseTest
+
+
+class TestEmailPasswordReset(BaseTest):
+
+    """Tests for all password reset and updating functionality."""
+
+    def test_successful_email_sending(self):
```

Comment:
In reply to the question about mocking the email sending process. In this test, i mocked the process in that. I signed up a user, send a post request to the `reset_password` url by calling this custom function `self.send_reset_password_email(self.email_forgot_password)` for the mail to be sent. Then I ascertained for the response